pkg/workflows/sdk/runtime.go

```diff
@@ -7,9 +7,29 @@ import (
 
 var BreakErr = capabilities.ErrStopExecution
 
+type Emitter interface {
+	// Emit sends a message with the given message and labels to the configured collector.
+	Emit(msg string, labels map[string]any) error
+}
+
+type EmitterFunc func(msg string, labels map[string]any) error
+
+func (f EmitterFunc) Emit(msg string, labels map[string]any) error {
+	return f(msg, labels)
+}
+
+type ContextEmitter interface {
```

Comment:
No point in supporting context on the guest-side, the wasmtime API isn't itself context-aware so it would be a pointless arg.

We should however support it on the host side by using the context passed into the execute call.